front-end/src/screens/UndoEmailChange/index.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useContext } from 'react';
+import { Segment, Header, Icon, Grid } from 'semantic-ui-react';
+import styled from 'styled-components';
+
+import { NotificationContext } from '../../context/NotificationContext';
+import { UserDetailsContext } from '../../context/UserDetailsContext';
+import { useUndoEmailChangeMutation } from '../../generated/graphql';
+import { useRouter } from '../../hooks';
+import { handleTokenChange } from '../../services/auth.service';
+import { NotificationStatus } from '../../types';
+import FilteredError from '../../ui-components/FilteredError';
+import Loader from '../../ui-components/Loader';
+
+interface Props {
+	className?: string
+}
+
+const UndoEmailChange = ({ className }:Props): JSX.Element => {
+	const router = useRouter();
+	const currentUser = useContext(UserDetailsContext);
+	const { queueNotification } = useContext(NotificationContext);
+	const [undoEmailChangeMutation, { error }] = useUndoEmailChangeMutation({
+		variables: {
+			token: router.query.token
+		}
+	});
+
+	useEffect(() => {
+		undoEmailChangeMutation().then(({ data }) => {
+			if (data && data.undoEmailChange && data.undoEmailChange.message && data.undoEmailChange.token) {
+				if (data.undoEmailChange.token) {
+					handleTokenChange(data.undoEmailChange.token);
+				}
+				currentUser.setUserDetailsContextState((prevState) => {
+					return {
+						...prevState,
+						email_verified: true
```

Comment:
I missed that previously but you need to set the email to the old one, and set `email_verified` to `false`

Reply:
done